### PR TITLE
ChatListHeader is now optional if there are no buttons or menu items injected, removed hardcoded "mark all as read" menu item

### PR DESCRIPTION
**Pull request by seekdavidlee**
Introduced the concept of `IChatListActions` that is injected in the buttons or menu items. This provides some externally expose functionality from `ChatList`.

### CODE REVIEW

File: samples/react-chat/src/App.tsx [hunk 1]
```diff
@@ -13,18 +13,35 @@ export const ChatAddIcon = (): JSX.Element => {
   return <ChatAddIconBundle color={iconColor} />;
 };
 
-const ChatListWrapper = memo(({ onSelected }: { onSelected: (e: GraphChatThread) => void }) => {
```
Comment (TechnicallyWilliams) [inline samples/react-chat/src/App.tsx:16]: Why remove the wrapper?

**Review by MattBillFred — COMMENTED** [review]
When I try the "Mark all as read" within the demo app, I'm not seeing the messages get marked as read.
These are the steps I'm doing:
1. yarn run build
2. yarn watch:chat

File: samples/react-chat/src/App.tsx (hunk 1 above)
Comment (seekdavidlee) [reply samples/react-chat/src/App.tsx:16]: I don't know if the wrapper provides any value, and I need easier access to `setShowNewChat(true)`

**Comment by seekdavidlee** [comment]
> When I try the "Mark all as read" within the demo app, I'm not seeing the messages get marked as read. These are the steps I'm doing:
> 
> 1. yarn run build
> 2. yarn watch:chat

Can you try `yarn; yarn build; yarn watch:chat`

**Comment by MattBillFred** [comment]
> > When I try the "Mark all as read" within the demo app, I'm not seeing the messages get marked as read. These are the steps I'm doing:
> > 
> > 1. yarn run build
> > 2. yarn watch:chat
> 
> Can you try `yarn; yarn build; yarn watch:chat`

I'm still seeing the same behavior

**Comment by MattBillFred** [comment]
> > > When I try the "Mark all as read" within the demo app, I'm not seeing the messages get marked as read. These are the steps I'm doing:
> > > 
> > > 1. yarn run build
> > > 2. yarn watch:chat
> > 
> > 
> > Can you try `yarn; yarn build; yarn watch:chat`
> 
> I'm still seeing the same behavior

Retested after new commit. The issue is resolved.

**Review by MattBillFred — APPROVED** [review]

**Review by TechnicallyWilliams — APPROVED** [review]

**Pull request merged**